components/automate-ui/src/app/services/projects-filter/projects-filter.service.ts

```diff
@@ -0,0 +1,76 @@
+import { Injectable } from '@angular/core';
+import { Observable, of as observableOf } from 'rxjs';
+import { Store } from '@ngrx/store';
+import { NgrxStateAtom } from 'app/ngrx.reducers';
+import { ProjectsFilterOption } from './projects-filter.reducer';
+import * as selectors from './projects-filter.selectors';
+import { LoadOptions, SaveOptions } from './projects-filter.actions';
+
+const STORE_OPTIONS_KEY = 'projectsFilter.options';
+
+@Injectable()
+export class ProjectsFilterService {
+  options$ = this.store.select(selectors.options);
+
+  selectionLabel$ = this.store.select(selectors.selectionLabel);
+
+  selectionCount$ = this.store.select(selectors.selectionCount);
+
+  selectionCountVisible$ = this.store.select(selectors.selectionCountVisible);
+
+  selectionCountActive$ = this.store.select(selectors.selectionCountActive);
+
+  constructor(private store: Store<NgrxStateAtom>) {}
+
+  loadOptions() {
+    this.store.dispatch(new LoadOptions());
+  }
+
+  saveOptions(options: ProjectsFilterOption[]) {
+    this.store.dispatch(new SaveOptions(options));
+  }
+
+  storeOptions(options: ProjectsFilterOption[]) {
+    localStorage.setItem(STORE_OPTIONS_KEY, JSON.stringify(options));
+  }
+
+  restoreOptions(): ProjectsFilterOption[] {
+    return JSON.parse(localStorage.getItem(STORE_OPTIONS_KEY));
+  }
+
+  fetchOptions(): Observable<ProjectsFilterOption[]> {
+    // Request some options available to the user.
+    return observableOf([
+      {
+        value: 'project-2',
+        label: 'Project 2',
+        checked: false
+      },
+      {
+        value: 'project-4',
+        label: 'Project 4',
+        checked: false
+      },
+      {
+        value: 'project-5',
+        label: 'Project 5',
+        checked: false
+      },
+      {
+        value: 'project-that-has-a-really-long-name',
+        label: 'Project that has a really long name',
+        checked: false
+      },
+      {
+        value: 'super-project',
+        label: 'Super Duper Project',
+        checked: false
+      }
+      // {
+      //   value: 'unassigned-resources',
+      //   label: 'Unassigned Resources',
+      //   checked: false
+      // }
+    ]);
+  }
```

Comment:
This is a bit different from the pattern used in `components/automate-ui/src/app/entities/*.requests.ts`. Curious what the motivation is to use a separate service vs when it is correct to add a new entity.

Reply:
It's just placeholder for async data; I'm not sure where projects available to the user will be loaded from or the shape they'll be received in. This can be replaced with a `*requests.ts` file or whatever else.

Reply:
> Curious what the motivation is to use a separate service vs when it is correct to add a new entity.

It's actually still unclear to me what we should be considering an 'entity' and modeled as such (things that live in a DB somewhere and/or come from an API?).

But as for the service itself, its main purpose is to keep all the ngrx bits out of components (and other parts of the system that may need access to this data: interceptors, other services, etc): https://medium.com/@thomasburlesonIA/ngrx-facades-better-state-management-82a04b9a1e39